mongorestore/mongorestore_test.go

```diff
@@ -107,12 +117,86 @@ func TestMongorestore(t *testing.T) {
 			restore.NSOptions.DB = "db1"
 			restore.InputReader = bsonFile
 			restore.TargetDirectory = "-"
+
+			result := restore.Restore()
+			So(result.Err, ShouldBeNil)
+			count, err := c1.CountDocuments(nil, bson.M{})
+			So(err, ShouldBeNil)
+			So(count, ShouldEqual, 100)
+
+			restore.NSOptions.Collection = ""
+			restore.NSOptions.DB = ""
+			restore.InputReader = nil
+			restore.TargetDirectory = ""
+			c1.Drop(nil)
+		})
+
+		Convey("and specifying an nsExclude option", func() {
+			restore.TargetDirectory = "testdata/testdirs"
+			restore.NSOptions.NSExclude = make([]string, 1)
+			restore.NSOptions.NSExclude[0] = "db1.c1"
+
 			result := restore.Restore()
 			So(result.Err, ShouldBeNil)
+			So(result.Successes, ShouldEqual, 0)
+			So(result.Failures, ShouldEqual, 0)
 
 			count, err := c1.CountDocuments(nil, bson.M{})
 			So(err, ShouldBeNil)
+			So(count, ShouldEqual, 0)
+
+			restore.TargetDirectory = ""
+			restore.NSOptions.NSExclude = nil
+			c1.Drop(nil)
+			c2.Drop(nil)
+			c3.Drop(nil)
+		})
+
+		Convey("and specifying an nsInclude option", func() {
+			restore.TargetDirectory = "testdata/testdirs"
+			restore.NSOptions.NSInclude = make([]string, 1)
+			restore.NSOptions.NSInclude[0] = "db1.c3"
+
+			result := restore.Restore()
+			So(result.Err, ShouldBeNil)
+			So(result.Successes, ShouldEqual, 0)
+			So(result.Failures, ShouldEqual, 0)
+
+			count, err := c3.CountDocuments(nil, bson.M{})
+			So(err, ShouldBeNil)
+			So(count, ShouldEqual, 0)
```

Comment:
Looking at it now, "nsIncluding" a collection with 0 documents might not be the best thing to do :P 

Maybe it's better to add a couple other collections to restore from if y'all think this isn't sufficient (meaning add new bson/metadata files to `testdata/testdirs/db1/`)

Reply:
I agree that restoring some documents would make this test a little more meaningful. Having no error and getting 0s for successes and failures doesn't indicate much.